pkg/externalLinks/ExternalLinksService.go

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2020 Devtron Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package externalLinks
+
+import (
+	"github.com/devtron-labs/devtron/internal/util"
+	"github.com/devtron-labs/devtron/pkg/sql"
+	"github.com/devtron-labs/devtron/pkg/user"
+	"go.uber.org/zap"
+	"time"
+)
+
+type ExternalLinksService interface {
+	Create(requests []*ExternalLinksRequest, userId int32) (*ExternalLinksCreateUpdateResponse, error)
+	GetAllActiveTools() ([]ExternalLinksMonitoringToolsRequest, error)
+	FetchAllActiveLinks(clusterIds int) ([]*ExternalLinksRequest, error)
+	Update(request *ExternalLinksRequest) (*ExternalLinksCreateUpdateResponse, error)
+	DeleteLink(id int, userId int32) (*ExternalLinksCreateUpdateResponse, error)
+}
+type ExternalLinksServiceImpl struct {
+	logger                          *zap.SugaredLogger
+	externalLinksToolsRepository    ExternalLinksToolsRepository
+	externalLinksClustersRepository ExternalLinksClustersRepository
+	externalLinksRepository         ExternalLinksRepository
+	userAuthService                 user.UserAuthService
+}
+type ExternalLinksMonitoringToolsRequest struct {
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+	Icon string `json:"icon"`
+}
+type ExternalLinksRequest struct {
+	Id               int       `json:"id"`
+	Name             string    `json:"name"`
+	Url              string    `json:"url"`
+	Active           bool      `json:"active"`
+	MonitoringToolId int       `json:"monitoringToolId"`
+	ClusterIds       []int     `json:"clusterIds"`
+	UpdatedOn        time.Time `json:"updatedOn"`
+	UserId           int32     `json:"-"`
+}
+
+type ExternalLinksCreateUpdateResponse struct {
```

Comment:
we are using this object as the response of deletion also. so define generic name. (ActionResponse)

Reply:
its ExternalLinkApiResponse now